### DISCUSSION TITLE
Bug in Search API Call?

### DISCUSSION THREAD

**Discussion by JoergBruenner** [DependencyTrack/dependency-track#2218] [Q&A]
Hi,

the following problem effects only  the API, both Swagger.json and scripted API calls. The GUI works as expected. I'm using DT 4.6.3. 

I've 2 Projects, called "a-a" and "a-b". I call the search API, to find project "a-c":
http://localhost:8081/api/v1/search/project?query=a-c

Result is:
{
  "results": {
    "project": [
      {
        "name": "a-a",
        "uuid": "9bf1cd3e-5ea8-4bfd-9c02-cb96ac37edba"
      },
      {
        "name": "a-b",
        "uuid": "369ef9d2-1cd9-4870-ba12-e8caa4347c9c"
      }
    ]
  }
}

**Comment by syalioune** [comment]
Hello @JoergBruenner 
This is indeed an issue in the search API.
Reproduced using a portfolio with projects "a-a", "a-b", "a-c" and searching for "a-e". The generated query for [lucene](https://lucene.apache.org/core/2_9_4/queryparsersyntax.html) is incorrect. See in the picture below

![index-search-issue](https://user-images.githubusercontent.com/6144741/205514193-cf24dcbb-6bb5-424b-ab35-93878ffde80d.png)

It would essentially search for project having "a" or "e" in the name because the search query is tokenized, thus projects "a-a", "a-b", "a-c" are returned.

**Reply by syalioune** [reply]
Issue https://github.com/DependencyTrack/dependency-track/issues/2233 created
